boot/bootutil/include/bootutil/bootutil.h

```diff
@@ -65,6 +65,38 @@ int boot_swap_type(void);
 int boot_set_pending(int permanent);
 int boot_set_confirmed(void);
 
+#define MCUBOOT_REQ_FLASH_MAP_SIZE          0
+#define MCUBOOT_REQ_FLASH_MAP_INFO          1
+
+struct mcuboot_api_itf {
+    uint32_t mcuboot_api_magic;
+    uint32_t mcuboot_version;
+    int (*mcuboot_ioctl)(int req, void *data, size_t len);
+};
+
+extern const struct mcuboot_api_itf mcuboot_api_vt;
+extern struct mcuboot_api_itf *p_mcuboot_api_vt;
+
+#if defined(__arm__)
+    #define MCUBOOT_API_INIT_(m)                                            \
+        extern const struct mcuboot_api_itf mcuboot_api_vt;                 \
+        asm("mov r4, %0\n\t"                                                \
+            "mov r5, %1"  :: "I" (m), "r" (&mcuboot_api_vt) : "r4", "r5");
```

Comment:
> The thing is the "clobbers" won't be used to prep and post the asm, but the compiler would be free to use those registers after, including for the call following it. It turns out that we're ok here because the call has no args, so the compiler feels no need to put anything else in the registers.

Well, as the ABI document I pointed before says: `r0-r3` are used for passing first 4 parameters, extra parameters go in the stack, so we're technically OK by using `r4-r5`

> But, I'd be more comfortable if it was explicit, either putting the jump in the asm itself (along with the arguments), or just using a regular call with arguments.

Yeah, I agree the calling in assembly is safer, I'll try that.

For completeness I tried to check the generated asm and confirm if it is done as defined in the previously mentioned ABI

```
int fn(int x0, int x1, int x2, int x3, int x4, int x5, int x6, int x7)
{
    int a = x0;
    int b = x1;
    return a == b;
}

int main(void)
{
    fn(0, 1, 2, 3, 4, 5, 6, 7);
    return 0;
}
```

`$ arm-none-eabi-gcc -mcpu=cortex-m4 -mthumb-interwork -mthumb -S test.c`

Relevant assembly (function call)

```
        ...
        movs	r3, #5
	str	r3, [sp, #4]
	movs	r3, #4
	str	r3, [sp]
	movs	r3, #3
	movs	r2, #2
	movs	r1, #1
	movs	r0, #0
	bl	fn
        ...
```

And it seems that `r4-r5` are never touched!